lib/internal/bootstrap_node.js

```diff
@@ -302,13 +302,13 @@
 
   function setupProcessFatal() {
 
-    process._fatalException = function(er) {
+    process._fatalException = function(er, fromPromise) {
       var caught;
 
       if (process.domain && process.domain._errorHandler)
         caught = process.domain._errorHandler(er) || caught;
 
-      if (!caught)
+      if (!caught && !fromPromise)
```

Comment:
@addaleax Mmmm, not quite though. This still calls `process.on('exit')`, which seems to be ok and makes logical sense.